File: app/src/main/java/com/enipro/data/remote/model/Experience.java
package com.enipro.data.remote.model;


import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.parceler.ParcelConstructor;

@org.parceler.Parcel
public class Experience {

    @SerializedName("_id")
    @Expose
    public ObjectId _id;

    @SerializedName("industry")
    @Expose
    public String industry;

    @SerializedName("organisation")
    @Expose
    public String organisation;

    @SerializedName("role")
    @Expose
    public String role;

    @SerializedName("from")
    @Expose
    public String from;

    @SerializedName("to")
    @Expose
    public String to;

    @ParcelConstructor
    public Experience(String industry, String organisation, String role, String from, String to) {
        this.industry = industry;
        this.organisation = organisation;
        this.role = role;
        this.from = from;
        this.to = to;
    }

    public ObjectId get_id() {
        return _id;
    }

    public void set_id(ObjectId _id) {
        this._id = _id;
    }

    public String getIndustry() {
        return industry;
    }

    public void setIndustry(String industry) {
        this.industry = industry;
    }

    public String getOrganisation() {
        return organisation;
    }

    public void setOrganisation(String organisation) {
        this.organisation = organisation;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public String getFrom() {
        return from;
    }

    public void setFrom(String from) {
        this.from = from;
    }

    public String getTo() {
        return to;
    }

    public void setTo(String to) {
        this.to = to;
    }
}

File: app/src/main/java/com/enipro/data/remote/model/Experience.kt
package com.enipro.data.remote.model


import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Experience
constructor(@field:SerializedName("industry")
            @field:Expose
            var industry: String, @field:SerializedName("organisation")
            @field:Expose
            var organisation: String, @field:SerializedName("role")
            @field:Expose
            var role: String, @field:SerializedName("from")
            @field:Expose
            var from: String, @field:SerializedName("to")
            @field:Expose
            var to: String) : Parcelable {

    @SerializedName("_id")
    @Expose
    var _id: ObjectId? = null

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString()) {
        _id = parcel.readParcelable(ObjectId::class.java.classLoader)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(industry)
        parcel.writeString(organisation)
        parcel.writeString(role)
        parcel.writeString(from)
        parcel.writeString(to)
        parcel.writeParcelable(_id, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Experience> {
        override fun createFromParcel(parcel: Parcel): Experience {
            return Experience(parcel)
        }

        override fun newArray(size: Int): Array<Experience?> {
            return arrayOfNulls(size)
        }
    }
}
